modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/web/internal/portlet/action/test/EditSegmentsEntryMVCRenderCommandTest.java

```diff
@@ -83,6 +94,65 @@ public void setUp() throws Exception {
 		_group = GroupTestUtil.addGroup();
 	}
 
+	@Test
+	public void testErrorMessage() throws Exception {
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequests();
+
+		MockLiferayPortletRenderResponse mockLiferayPortletRenderResponse =
+			new MockLiferayPortletRenderResponse();
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			WebKeys.USER, TestPropsValues.getUser());
+
+		_expandoTable = ExpandoTestUtil.addTable(
+			PortalUtil.getClassNameId(User.class), "CUSTOM_FIELDS");
+
+		ExpandoColumn expandoColumn = _addExpandoColumn(
+			_expandoTable, RandomTestUtil.randomString(),
+			ExpandoColumnConstants.BOOLEAN,
+			ExpandoColumnConstants.INDEX_TYPE_KEYWORD);
+
+		SegmentsEntry segmentsEntry = _addSegmentEntry(
+			String.format(
+				"(customField/%s eq %s)", _encodeName(expandoColumn),
+				StringPool.TRUE));
+
+		_expandoColumnLocalService.deleteColumn(expandoColumn);
+
+		mockLiferayPortletRenderRequest.setParameter(
+			"segmentsEntryId",
+			String.valueOf(segmentsEntry.getSegmentsEntryId()));
+
+		_mvcRenderCommand.render(
+			mockLiferayPortletRenderRequest, mockLiferayPortletRenderResponse);
+
+		try {
+			Map<String, Object> data = ReflectionTestUtil.invoke(
+				mockLiferayPortletRenderRequest.getAttribute(
+					"EDIT_SEGMENTS_ENTRY_DISPLAY_CONTEXT"),
+				"getData", new Class<?>[0]);
+
+			Map<String, Object> props = (Map<String, Object>)data.get("props");
+
+			JSONArray contributorsJSONArray = (JSONArray)props.get(
+				"contributors");
+
+			JSONObject initialQueryJSONObject =
+				contributorsJSONArray.getJSONObject(
+					0
+				).getJSONObject(
+					"initialQuery"
+				);
+
+			Assert.assertEquals(
+				"true", initialQueryJSONObject.getString("error"));
+		}
+		catch (Exception exception) {
+			Assert.assertNotNull(exception);
```

Comment:
What is the exception you expect, you should add it as expected in the @Test annotation.

Reply:
Ok, thank you

Reply:
@cgoncas I have a doubt about this. Initially I used the try-catch form because it handles the cases when the FF is true or false, and the test pass in both cases. In case I use @test annotation, the test will always expect to throw an exception, so when the FF is true the test won't pass. What do you think, should I use the @test notation even if the test would fail with the FF=true and change the test after we remove the FF?